cpp/src/decisiontree/kernels/gini_def.h

```diff
@@ -19,9 +19,11 @@
 #include "cub/cub.cuh"
 #include "../memory.cuh"
 #include <vector>
+#include "cuda_utils.h"
+#include <math.h>
 
 template<class T>
-struct GiniQuestion {
+struct MetricQuestion {
```

Comment:
Maybe better to rename this header file in light of this change to `metric_def.h`?

Reply:
agreed. Will change it.